libc/src/locale/locale.h

```diff
@@ -0,0 +1,36 @@
+//===-- Implementation header for the locale --------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_LIBC_SRC_LOCALE_LOCALECONV_H
+#define LLVM_LIBC_SRC_LOCALE_LOCALECONV_H
+
+#include "src/__support/macros/attributes.h"
+#include "src/__support/macros/config.h"
+
+#include "include/llvm-libc-types/locale_t.h"
+
+#include <stddef.h>
+
+namespace LIBC_NAMESPACE_DECL {
+
+// We only support the "C" locale right now.
+static constexpr size_t MAX_LOCALE_NAME_SIZE = 2;
+
+struct __locale_data {
+  char name[MAX_LOCALE_NAME_SIZE];
+};
+
+// The pointer to the default "C" locale.
+extern __locale_t c_locale;
+
+// The global locale instance.
+LIBC_THREAD_LOCAL extern locale_t locale;
```

Comment:
This isn't going to work on baremetal where thread local variables generally aren't available. According to the documentation the locale functions aren't expected to be thread-safe so this shouldn't be required. Can we make it a global variable?

Reply:
Isn't that what `LIBC_THREAD_LOCAL` does for unsupported platforms?

Reply:
That only covers the GPU build at the moment: https://github.com/llvm/llvm-project/blob/c975dc1da03d684604ddf787b07b63fb8e903648/libc/src/__support/macros/attributes.h#L31

Reply:
We currently don't have any define for the baremetal target to conditionalize the build on although I suspect we'll need one sooner or later.

Reply:
I'd still prefer omitting the `LIBC_THREAD_LOCAL` for now and add it later if it turns out to be needed on some platforms.

Reply:
Done.